lib/core/src/system/gpu/data/texture.rs

```diff
@@ -427,31 +509,32 @@ pub struct BoundTexture<I,T> {
 /// Texture bound to GL context.
 #[derive(Debug)]
 pub struct BoundTextureData<I,T> {
-    texture    : Texture<I,T>,
     gl_texture : WebGlTexture,
     context    : Context,
+    format     : PhantomData<I>,
+    elem_type  : PhantomData<T>,
 }
```

Comment:
First of all the name `format` is misleading. This is `internal_format`, not `format. Second of all, I would keep the names more visible. Why not this instead?

```rust
pub struct BoundTextureData<InternalFormat,ElemType> {
    gl_texture : WebGlTexture,
    context    : Context,
    phantom    : PhantomData2<InternalFormat,ElemType>,
}
```

Reply:
There was `I` and `T` before, here and in many other places, so I though that is your code style :)